tests/pytest/oauth/test_views.py

```diff
@@ -0,0 +1,56 @@
+from django.http import HttpResponse
+from django.urls import reverse
+
+from benefits.core import session
+from benefits.oauth.views import logout, post_logout, _generate_redirect_uri
+
+import pytest
+
+
+@pytest.mark.request_path("/oauth/logout")
+def test_logout(mocker, session_request):
+    # logout internally calls _deauthorize_redirect
+    # this mocks that function and a success response
+    # and returns a spy object we can use to validate calls
+    message = "logout successful"
+    spy = mocker.patch("benefits.oauth.views._deauthorize_redirect", return_value=HttpResponse(message))
+
+    token = "token"
+    session.update(session_request, oauth_token=token)
+    assert session.oauth_token(session_request) == token
+
+    result = logout(session_request)
+
+    spy.assert_called_with(token, "https://testserver/oauth/post_logout")
+    assert result.status_code == 200
+    assert message in str(result.content)
+    assert session.oauth_token(session_request) is False
+
+
+@pytest.mark.request_path("/oauth/post_logout")
+def test_post_logout(session_request):
+    origin = reverse("core:index")
+    session.update(session_request, origin=origin)
+
+    result = post_logout(session_request)
+
+    assert result.status_code == 302
+    assert result.url == origin
```

Comment:
If the intention is to keep the user logged into Benefits, we should confirm that in this test. Also, let's add a test for no `origin`.

Reply:
But... that is not the intention? This is the logout code, the user is signing out of everything. 

> This PR adds a new route /oauth/logout that signs the user out of all 3 environments: Login.gov, the auth server, and the Django app.

Reply:
Sorry, this was tied to my misunderstand of where/whether the `oauth_token` was cleared.

Reply:
Just thought, something to add to this test - the user should already be logged out of their Django session. That would have made this more clear from the beginning, apologies! Adding...